src/improvements/tasks/StateOverrideManager.sol

```diff
@@ -154,8 +154,21 @@ abstract contract StateOverrideManager is CommonBase {
         Simulation.StateOverride[] memory parsedOverrides = new Simulation.StateOverride[](targetAddresses.length);
         for (uint256 i = 0; i < targetAddresses.length; i++) {
             string memory overridesPath = string.concat(stateOverridesKey, ".", targetStrings[i]);
+            bytes memory tomlOverrides = vm.parseToml(toml, overridesPath);
             Simulation.StorageOverride[] memory storageOverrides =
-                abi.decode(vm.parseToml(toml, overridesPath), (Simulation.StorageOverride[]));
+                abi.decode(tomlOverrides, (Simulation.StorageOverride[]));
+
+            // Reencode the overrides back to bytes and ensure that the roundtrip encoding is the same as the original.
+            // This is a hacky form of type safety to make up for the lack of it in the toml parser.
+            bytes memory reencoded = abi.encode(storageOverrides);
+            require(
+                reencoded.length == tomlOverrides.length,
+                "StateOverrideManager: Failed to reencode overrides, ensure all values are encoded as bytes32 strings"
+            );
+            require(
+                keccak256(reencoded) == keccak256(tomlOverrides),
+                "StateOverrideManager: Failed to reencode overrides, ensure all values are encoded as bytes32 strings"
```

Comment:
I feel like defaulting to decimal numbers is more readable, what do you think of this
```suggestion
                "StateOverrideManager: Failed to reencode overrides, ensure any decimal numbers are not in quotes"
```